torch/csrc/generic/StorageMethods.cpp

```diff
@@ -41,7 +60,7 @@ static PyObject * THPStorage_(isPinned)(PyObject *_self, PyObject *noargs)
   HANDLE_TH_ERRORS
   auto self = (THPStorage*)_self;
 #if defined(USE_CUDA)
-  return PyBool_FromLong(at::globalContext().isPinnedPtr(THWStorage_(data)(LIBRARY_STATE self->cdata)));
+  return PyBool_FromLong(at::globalContext().isPinnedPtr(reinterpret_cast<c10::StorageImpl*>(self->cdata)->data<scalar_t>()));
```

Comment:
Why do you need `reinterpret_cast` for `cdata` but not other uses of `THWStorage`?

Reply:
Right, these casts aren't needed